jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/transport/HttpDestination.java

```diff
@@ -489,15 +484,22 @@ public void release(Connection connection)
         else
         {
             if (LOG.isDebugEnabled())
-                LOG.debug("{} is stopped", client);
+                LOG.debug("Closing {}, {} is stopped", connection, client);
             connection.close();
         }
     }
 
+    private boolean releaseConnection(Connection connection)
```

Comment:
Two things confusing about this method on first glance:\

 + how is `releaseConnection(Connection)` different to `releaseConnection(Connection)`? Some better names and/or javadoc would help.
 + why is the connection closed if it is not released?  If that is an unexpected situation, then a warn or debug would be good? A comment in the code to explain would be good.    I'm assuming that a connection can be acquired multiple times due to multiplexing, so releasing to non-zero should be normal?  or does the boolean return mean something else?   Oh the boolean return from the release is special.

Maybe call this `releaseOrClose(Connection)`, because either the connection is released or it is closed?